pkg/alertmanager/api.go

```diff
@@ -256,7 +256,7 @@ func validateUserConfig(logger log.Logger, cfg alertspb.AlertConfigDesc, limits
 		templateFiles[i] = filepath.Join(userTempDir, t)
 	}
 
-	_, err = template.FromGlobs(templateFiles...)
+	_, err = template.FromGlobs(templateFiles)
```

Comment:
I think we need to inject the tenant ID function there as well, otherwise a template using the function may fail.  Can you verify it adding a unit test too, please? You can add a test case to `TestAMConfigValidationAPI`.

Reply:
Thanks, fixed in https://github.com/grafana/mimir/pull/3758/commits/975295f6c74afad4ef5d2d47b8f68ba8fd34f48e